src/elements/common/annotator-context/withAnnotations.tsx

```diff
@@ -0,0 +1,47 @@
+import * as React from 'react';
+import AnnotatorContext from './AnnotatorContext';
+import { Action, AnnotationActionEvent, AnnotatorState, Status } from './types';
+
+type WrapperProps = {
+    showAnnotationControls: boolean;
+};
+
+type WrapperReturn<P> = React.ComponentClass<P & WrapperProps>;
+
+export default function withAnnotations<P extends object>(
+    WrappedComponent: React.ComponentType<P & WrapperProps>,
+): WrapperReturn<P> {
+    class ComponentWithAnnotations extends React.Component<P & WrapperProps, AnnotatorState> {
+        static displayName: string;
+
+        state: AnnotatorState = {
+            annotation: null,
```

Comment:
How will this work when multiple operations are performed in sequence? Will this state get copied elsewhere? Any chance of conflicts between events?

Reply:
As discussed, I think a sequence of operations should be okay. Each discrete event would trigger a value change in the AnnotatorContext which would cause a re-render of any component using the `withAnnotatorContext` hoc. So even if the API to create an annotation took a while, we would see a sequence like: pending create 1, pending create 2, successful create 1, successful create 2, and each state change can be handled by the consuming component. 

Something to look out for is if somehow the events happened so fast that `setState` batches the change together which would result in a loss of data. That happening for a create or update should be fine since missing out on the pending state isn't a big deal but the chances of that happening for creating two annotations back to back seems low